### PR TITLE
Feature basic obstacles

### PR DESCRIPTION
**Pull request by marcelkuipers**
This PR implements a basic way to implement obstacles and generate them into the level.
Please notice the point of  this PR is not to implement dynamic spawning behaviour. In other words, the DamageDealerClass will need to be reworked in a future PR.

### CODE REVIEW

**Comment by marcelkuipers** [comment]
Closed because of failing build

**Comment by marcelkuipers** [comment]
This PR is now up-to-date with the refactored environment.

File: ContextDesktop/src/main/java/com/github/migi_1/Context/damageDealers/DamageDealerGenerator.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+     * Instantiate local variables.
+     * @param commander The Commander object.
+     */
+    public DamageDealerGenerator(Commander commander) {
+        this.location = commander.getModel().getLocalTranslation();
+        this.commander = commander;
+        this.damageDealerFactory = new StaticObstacleFactory();
+        this.damageDealerList = new ArrayList<DamageDealer>();
+    }
+
```
Comment (dVoorhout) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/damageDealers/DamageDealerGenerator.java:41]: I guess this is an outdated comment, please fix

File: ContextDesktop/src/main/java/com/github/migi_1/Context/model/MainEnvironment.java [hunk 2]
```diff
@@ -84,9 +98,26 @@ public void initialize(AppStateManager stateManager, Application app) {
     public void update(float tpf) {
     	super.update(tpf);
 
+        checkCollision();
     	updateTestWorld();
     }
 
+    private void checkCollision() {
```
Comment (NilsHullegien) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/MainEnvironment.java:108]: Missing javadoc

File: ContextDesktop/src/main/java/com/github/migi_1/Context/model/MainEnvironment.java [hunk 3]
```diff
@@ -84,9 +98,26 @@ public void initialize(AppStateManager stateManager, Application app) {
...
+        //if a collision takes place, remove the colliding object and slow down
+        if (results.size() > 0) {
+            getRootNode().detachChild(damageDealerGenerator.removeDamageDealer().getModel());
+            results = new CollisionResults();
+            ((ConstantSpeedMoveBehaviour) platform.getMoveBehaviour()).collided();
+            ((ConstantSpeedMoveBehaviour) commander.getMoveBehaviour()).collided();
+        }
+    }
+
     /**
```
Comment (dVoorhout) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/MainEnvironment.java:124]: javadoc missing

Comment (dVoorhout) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/MainEnvironment.java:124]: me and Nils are on the same page apparently ;p

**Comment by NilsHullegien** [comment]
Looks good now

**Pull request merged**